Header header
float32[] curr_state
float32[] next_state